_posts/2021-01-01-getting-started-with-graphql.md

```diff
@@ -0,0 +1,288 @@
+```
+Here we send a request for fetching 10 recent purchases with the fields title, and description in each record.
+
+To avoid making the low-level HTTP calls, we should use a GraphQL client as an abstraction layer to take care of:
+1. sending the request and handling the response
+2. view layer integrations and optimistic UI updates
+3. caching query results
+
+There are several [client frameworks](https://graphql.org/code/#javascript-client) available with popular ones being [Apollo Client](https://github.com/apollographql/apollo-client), [Relay (from Facebook)](https://facebook.github.io/relay/), and [urql](https://github.com/FormidableLabs/urql).
+
+## GraphQL Example with Spring Boot
+
+We will use a Spring Boot application to build a GraphQL server implementation. For this, let us first create a Spring Boot application with the [Spring Initializr](https://start.spring.io/#!type=maven-project&language=java&platformVersion=2.4.2.RELEASE&packaging=jar&jvmVersion=11&groupId=io.pratik&artifactId=graphqldemo&name=graphqldemo&description=Demo%20project%20for%20GraphQL%20with%20Spring%20Boot&packageName=io.pratik.graphqldemo&dependencies=web,lombok,h2). 
+
+## Adding the Dependencies
+For GraphQL server, we will add the following Maven dependencies: 
+
+```xml
+    <dependency>
+      <groupId>com.graphql-java</groupId>
+      <artifactId>graphql-spring-boot-starter</artifactId>
+      <version>5.0.2</version>
+    </dependency>
+    <dependency>
+      <groupId>com.graphql-java</groupId>
+      <artifactId>graphql-java-tools</artifactId>
+      <version>5.2.4</version>
+    </dependency>
+```
+Here we have added `graphql-spring-boot-starter` as a GraphQL starter and a Java tools module `graphql-java-tools`.
+
+### Defining the GraphQL Schema 
+We can either take a top-down approach by defining the schema and then create POJO for each type or a bottom-up approach by creating the POJOs first followed by schema from the POJO classes. 
+
+The GraphQL schema needs to be defined in a file with the extension `graphqls`. Let us define our schema in a file `product.graphqls`:
+
+```
+type Product {
+    id: ID!
+    title: String!
+    description: String!
+    category: String
+    madeBy: Manufacturer!
+}
+
+type Manufacturer {
+    id: ID!
+    name: String!
+    address: String
+}
+
+# The Root Query for the application
+type Query {
+    myRecentPurchases(count: Int, customerID: String): [Product]!
+    lastVisitedProducts(count: Int, customerID: String): [Product]!
+    productsByCategory(category: String): [Product]!
+}
+
+# The Root Mutation for the application
+type Mutation {
+    addRecentProduct(title: String!, description: String!, category: String) : Product!
+}
+
+```
+Here we have added three operations to our Query and a Mutation for adding recent products.
+
+Next, we define the POJO classes for the Object types: `Product` and `Manufacturer`:
+
+```java
+@Data
```

Comment:
Maybe explain this annotation in a sentence (or remove it from the code snippet if it's not needed in the context of this article).